src/shader_recompiler/frontend/translate/scalar_alu.cpp

```diff
@@ -154,6 +206,30 @@ void Translator::S_CMP(ConditionOp cond, bool is_signed, const GcnInst& inst) {
     ir.SetScc(result);
 }
 
+void Translator::S_CMPK(ConditionOp cond, bool is_signed, const GcnInst& inst) {
+    const s32 simm16 = inst.control.sopk.simm;
+    const IR::U32 lhs = GetSrc(inst.src[0]);
```

Comment:
should be GetSrc(inst.dst[0]), src[0] is undefined

see src/shader_recompiler/frontend/decode.cpp
```c++
void GcnDecodeContext::decodeInstructionSOPK(u32 hexInstruction) {
    u32 sdst = bit::extract(hexInstruction, 22, 16);
    u32 op = bit::extract(hexInstruction, 27, 23);

    m_instruction.opcode = static_cast<Opcode>(op + static_cast<u32>(OpcodeMap::OP_MAP_SOPK));

    m_instruction.dst[0].field = getOperandField(sdst);
    m_instruction.dst[0].code = sdst;
    m_instruction.dst_count = 1;

    m_instruction.control.sopk = *reinterpret_cast<InstControlSOPK*>(&hexInstruction);
}
```

Reply:
Thanks for the heads ups!